cpp/tests/fixed_point/fixed_point_tests.cpp

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
I think this PR also needs a fixed_point_tests.cu with kernels and/or thrust calls that exercise the type in device code and ensures results are correct.

If it's not tested, it doesn't work! (And I can guarantee at this point it doesn't work because there are no device-callable constructors.)

Reply:
Done (there seems to be a problem with "map accumulator algorithms i.e. `*_scan` and `adjacent_difference`). Will follow up later. `reduce` and `transform` working though.